pkg/health/health.go

```diff
@@ -43,6 +43,9 @@ func (httpProbe HTTPProbe) Probe() (int, error) {
 	client := &http.Client{}
 
 	if httpProbe.Protocol == ProtocolHTTPS {
+		// Certificate validation is to be skipped for HTTPS probes
+		// similar to how k8s api server handles HTTPS probes.
+		// #nosec G402
```

Comment:
I appreciate the explanations for each override in the commit messages. I think a comment like this right next to the override is also good just to shortcut having to `git blame` to see why the override is there. golangci-lint also has lots of ways to configure overrides in its config file if that's easier than tagging individual rule violations: https://golangci-lint.run/usage/configuration/. Each of these seem isolated enough that that might not be useful for these, but worth knowing about in any case.

Reply:
These are the very few individual false positives that are better off explicitly annotating. There is one more that I need to clean up G104 in a subsequent change, but its isn't a problem as of now due to the linter running only on new changes.

I also found it to be rather too verbose to explain each override inline in the code, and it can be rather distracting from a linting perspective.